avocado/core/job.py

```diff
@@ -275,6 +275,9 @@ def _make_test_suite(self, references=None):
         loader.loader.load_plugins(self.args)
         try:
             suite = loader.loader.discover(references)
```

Comment:
Well I was looking forward to the API change. I imagined something like `key-variable` interface. Something like `{"name": "Passtest:test", "tags": ["fast", "-disk"]}` and I hope we'll get to that stage eventually (not necessarily in the first implementation, the format is implementation detail and the results should look the same).

About speed, the filter-by-tags is only available to instrumental tests and loaders can make note of that only filtering those tests (no `if not instrumented...`). Then it won't include the tests in test_suite, instead of creating big `test_suite` and then picking some of them into `filtered_test_suite` so I see a potential for optimizations there. Anyway as I said before, it can be addressed later, but I'd like to see that coming as we do several rounds of filtering on a different places.